### ISSUE TITLE
[java] WhileLoopWithLiteralBoolean misses constant expression

### ISSUE THREAD

**Issue by Jane-PL-256** [pmd/pmd#3370]
**Affects PMD Version:6.35.0** 

**Rule:WhileLoopWithLiteralBoolean**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.36.0/pmd_rules_java_bestpractices.html#whileloopwithliteralboolean>

**Description:**
PMD should have reported a warning to detect the bug because the conditional expression is actually a constant, the program will exist a useless while-loop.

**Code Sample demonstrating the issue:**

```java
public void whileLoopWithLiteralBoolean() {
  boolean var = true;
  do {
    System.out.println("test");
  } while(false | false);  // should report a warning here
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Belle-PL** [comment]
@oowekyala, Hi Clément, could you please kindly help check this issue? Thanks.

**Comment by oowekyala** [comment]
@Belle-PL Hi, would you like to tackle this issue and need guidance?
A simple implementation strategy for a fix would be to check that the expression only contains literals and boolean operators. Then, it would be a constant expression. 

I think this bug has limited impact and I won't be working on this myself before pmd 7. But contributions are welcome on master (pmd 6) :)

**Comment by Belle-PL** [comment]
@oowekyala Hi, thanks for your kind reply. I will try to fix this issue, please give me some time to read the code. Thanks.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
